src/provider/dynamodb.ts

```diff
@@ -0,0 +1,198 @@
+'use strict';
+
+import { DynamoDB } from 'aws-sdk';
+import AWS = require('aws-sdk');
+import { DocumentClient, ItemList } from 'aws-sdk/clients/dynamodb';
+import { AWSConfig } from '../aws/config';
+import { Event } from '../model/event';
+import { Stream } from '../model/stream';
+import { PersistenceProvider } from './provider';
+
+
+/**
+ * A Persistence Provider that handle all the data in Dynamodb.
+ */
+export class DynamodbProvider implements PersistenceProvider {
+    private documentClient: DocumentClient;
+    private dynamoDB: DynamoDB;
+    private initialized = false;
+
+    constructor(awsConfig: AWSConfig) {
+        AWS.config.update(awsConfig);
+
+        this.documentClient = new DynamoDB.DocumentClient();
+        this.dynamoDB = new AWS.DynamoDB();
+    }
+
+    public async addEvent(stream: Stream, data: any) {
+        await this.ensureTables();
```

Comment:
Is it possible to move `ensureTables` to the constructor? This way it'll be called a single time.